src/store.js

```diff
@@ -1,94 +1,180 @@
 import Vue from 'vue'
-import { $set } from './helpers'
-import { loadState, dumpState } from './storage'
-import { subscribe } from './watcher'
+import { namespaceSeparator, getModuleByNamespace } from './namespace'
+import { registerStorage } from './storage'
+import { setWatcher, subscribe } from './watcher'
+import { callIfFunction } from './utils'
 
 export function createStore ({
-  ctx,
   state,
+  hydrate,
   actions,
   getters,
-  hydrate,
-  localStorage,
-  sessionStorage
-}) {
-  const initialState = state(ctx)
+  modules,
+  storage
+}, vm) {
+  if (vm) {
+    setWatcher(vm)
+  }
+
+  let initialState = state ? callIfFunction(state) : {}
 
-  if (ctx._isVue) {
-    ctx = {}
+  if (hydrate) {
+    initialState = hydrate(initialState)
   }
 
-  if (localStorage) {
-    for (const namespace of localStorage) {
-      loadState(window.localStorage, state, data => $set(ctx.$state, namespace, data))
-      subscribe(ctx, namespace, snapshot => dumpState(window.localStorage, namespace, snapshot))
+  const store = {
+    $state: Vue.observable(initialState),
+    $getters: {},
+    $actions: {},
+    subscribe: (...args) => subscribe(store, ...args),
```

Comment:
These are some helper methods on the inject `this.$stator` prop. Shouldnt be hurting initial store creating performance that much.